connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -437,6 +439,38 @@ public Void call() throws Exception {
         );
     }
 
+    private ConnectorInfo createConnectorInfo(String connector) {
+        if (!configState.contains(connector))
+            return null;
+        Map<String, String> config = configState.rawConnectorConfig(connector);
+        return new ConnectorInfo(connector, config, configState.tasks(connector),
+                                 connectorTypeForClass(config.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG)));
+    }
+
+
+    @Override
+    public void connectorsStatus(Callback<Map<String, ConnectorStateInfo>> callback) {
+        log.trace("Submitting connectors info request");
+
+        addRequest(
+            new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    if (checkRebalanceNeeded(callback))
+                        return null;
+
+                    final Map<String, ConnectorStateInfo> out = new HashMap<>();
+                    for (String connector : configState.connectors()) {
+                        out.put(connector, connectorStatus(connector));
```

Comment:
The `connectorStatus(connector)` call might fail with a `NotFoundException` if the connector was removed after the `configState.connector()` method is called but before the status for the removed connector is asked for. Although this shouldn't happen within the process (since requests are handled sequentially by a single thread), it may be possible that this herder is not the leader, that the leader performed the change, and that this herder's config state backing store read that change after the `configState.connector()` method was called but before the `connectorStatus(connector)` method is called for that connector.

Should be easily handled with a try-catch, and if the connector with the specified name is not found then simply continue to the next connector name. Something like:
```suggestion
                        try {
                            out.put(connector, connectorStatus(connector));
                        } catch (NotFoundException e) {
                            // do nothing with connectors that were just removed
                        }
```

Note that if a connector is *added* with similar timing, the new connector name will not be returned from `configState.connectors()` and the new connector will not be included in the results. I think that's fine, considering the result of this method call would still be consistent with the state at the time the `configState.connectors()` call is made. Call it again, and you'd see the new connector.